src/core/net/nd6.hpp

```diff
@@ -42,22 +42,26 @@
 #include <stdint.h>
 
 #include <openthread/netdata.h>
+#include <openthread/platform/infra_if.h>
 #include <openthread/platform/toolchain.h>
 
+#include "border_router/infra_if.hpp"
 #include "common/const_cast.hpp"
 #include "common/encoding.hpp"
 #include "common/equatable.hpp"
 #include "common/heap_array.hpp"
 #include "net/icmp6.hpp"
 #include "net/ip6.hpp"
+#include "net/ip6_headers.hpp"
 #include "thread/network_data_types.hpp"
 
 namespace ot {
 namespace Ip6 {
 namespace Nd {
 
 typedef NetworkData::RoutePreference RoutePreference; ///< Route Preference
-
+typedef Data<kWithUint16Length>      Icmp6Packet;     ///< A data buffer containing an ICMPv6 packet.
+using LinkLayerAddress = ot::BorderRouter::InfraIf::LinkLayerAddress;
```

Comment:
Maybe using `typedef` here for consistency with above definitions?
Also adding new line after the last def and before `/*` ?

```suggestion
typedef Data<kWithUint16Length>                 Icmp6Packet;      ///< A data buffer for an ICMPv6 packet.
typedef BorderRouter::InfraIf::LinkLayerAddress LinkLayerAddress; ///< An infra-if link-layer address.

```